test/spoof/error_checks.go

```diff
@@ -0,0 +1,36 @@
+package spoof
+
+import (
+	"net"
+	"net/url"
+	"strings"
+)
+
+func isTCPTimeout(err error) bool {
+	if err, ok := err.(net.Error); ok && err.Timeout() {
+		return true
+	}
+	return false
+}
+
+func isDNSError(err error) bool {
+	if err, ok := err.(*url.Error); err != nil && ok {
+		if err, ok := err.Err.(*net.OpError); err != nil && ok {
+			if err, ok := err.Err.(*net.DNSError); err != nil && ok {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func isTCPConnectRefuse(err error) bool {
```

Comment:
this is easy to test, by connecting to a port/target that won't accept.

Reply:
Done